demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -1,1081 +1,557 @@
+                                      uint32_t ulPropertyVersion,
+                                      bool * pxOutMaxTempChanged )
+{
+    *pxOutMaxTempChanged = false;
+    xDeviceCurrentTemperature = xNewTemperatureValue;
+
+    /* Update maximum or minimum temperatures. */
+    if( xDeviceCurrentTemperature > xDeviceMaximumTemperature )
+    {
+        xDeviceMaximumTemperature = xDeviceCurrentTemperature;
+        *pxOutMaxTempChanged = true;
+    }
+    else if( xDeviceCurrentTemperature < xDeviceMinimumTemperature )
+    {
+        xDeviceMinimumTemperature = xDeviceCurrentTemperature;
+    }
+
+    /* Calculate the new average temperature. */
+    ulDeviceTemperatureCount++;
+    xDeviceTemperatureSummation += xDeviceCurrentTemperature;
+    xDeviceAverageTemperature = xDeviceTemperatureSummation / ulDeviceTemperatureCount;
+
+    LogInfo( ( "Client updated desired temperature variables locally." ) );
+    LogInfo( ( "Current Temperature: %2f", xDeviceCurrentTemperature ) );
+    LogInfo( ( "Maximum Temperature: %2f", xDeviceMaximumTemperature ) );
+    LogInfo( ( "Minimum Temperature: %2f", xDeviceMinimumTemperature ) );
+    LogInfo( ( "Average Temperature: %2f", xDeviceAverageTemperature ) );
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Gets the reported properties payload with the maximum temperature value.
+ */
+static uint32_t prvGetNewMaxTemp( double xUpdatedTemperature, 
+                              uint8_t * ucReportedPropertyPayloadBuffer,
+                              uint32_t ulReportedPropertyPayloadBufferSize )
+{
+    AzureIoTResult_t xResult;
+    AzureIoTJSONWriter_t xWriter;
+    int32_t lBytesWritten;
+
+    /* Initialize the JSON writer with the buffer to which we will write the payload with the new temperature. */
+    xResult = AzureIoTJSONWriter_Init( &xWriter, ucReportedPropertyPayloadBuffer, ulReportedPropertyPayloadBufferSize );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTJSONWriter_AppendBeginObject( &xWriter );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTJSONWriter_AppendPropertyName( &xWriter, ( const uint8_t * ) sampleazureiotPROPERTY_MAX_TEMPERATURE_TEXT,
+                                                     sizeof( sampleazureiotPROPERTY_MAX_TEMPERATURE_TEXT ) - 1 );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTJSONWriter_AppendDouble( &xWriter, xUpdatedTemperature, sampleazureiotDOUBLE_DECIMAL_PLACE_DIGITS );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTJSONWriter_AppendEndObject( &xWriter );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    lBytesWritten = AzureIoTJSONWriter_GetBytesUsed( &xWriter );
+
+    return lBytesWritten;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Send updated maximum temperature value to IoT Hub.
+ */
+static void prvSendNewMaxTemp( double xUpdatedTemperature )

```

Comment:
>prvSendNewMaxTemp [](http://example.com/codeflow?start=12&length=17)

Can we better name this, as now this is not sending, but more like preparing the payload